narwhals/_pandas_like/expr.py

```diff
@@ -388,11 +388,14 @@ def alias(self, name: str) -> Self:
 
     def over(self, keys: list[str]) -> Self:
         cumulative_functions_to_pandas_equivalent = {
-            "col->cum_count": "cumcount",
             "col->cum_sum": "cumsum",
             "col->cum_min": "cummin",
             "col->cum_max": "cummax",
             "col->cum_prod": "cumprod",
+            # Pandas cumcount starts counting from 0 while Polars starts from 1
+            # Pandas cumcount counts nulls while Polars does not
+            # So, instead of using "cumcount" we use "cumsum" on notna() to get the same result
+            "col->cum_count": lambda s: s.notna().cumsum(),
```

Comment:
we can't use a lambda function here, it'll be slow

Reply:
True, it's going to be definetly slower than using a transformation passed by string. I am not sure we can do much better tho. I will try a couple of alternatives and I will be back to you.

```python
data = {
    "a": ["a", "a", "b", "b", "b"] * 1_000_000,
    "b": [1, 2, 3, None, 3] * 1_000_000,
}

df = pd.DataFrame(data)

%timeit df.groupby("a").transform("cumsum")
# 134 ms ± 1.29 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit df.groupby("a").transform(lambda s: s.cumsum())
# 448 ms ± 2.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

%timeit df.groupby("a").transform(pd.Series.cumsum)
# 446 ms ± 455 μs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
That's what I got now

```python
%timeit df.assign(b=df["b"].notna()).groupby("a")["b"].transform("cumsum")
# 199 ms ± 4.14 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Thanks for challenging it. Do you think it can go any faster?

Reply:
nice - even better would probably to use `concat` with `copy=False`, look at `with_columns` in `_pandas_like`

Reply:
How about using `with_columns` directly :)

https://github.com/ClaudioSalvatoreArcidiacono/narwhals/blob/fe99682f106bc841b3b627a85c674b4d28952075/narwhals/_pandas_like/expr.py#L418-L419

Reply:
yup, great one!